### PR TITLE
CFE-2263 Add 'select_end_match_eof' documentation.

### PR DESCRIPTION
**Pull request by pasinskim**

### CODE REVIEW

**Comment by pasinskim** [comment]
@estenberg @nickanderson can you check the wording here?

File: reference/promise-types/edit_line.markdown [hunk 1]
```diff
@@ -411,7 +412,37 @@ mark out the region of a file to be edited.
...
+```cf3
+     body select_region example(x)
+     {
+     select_start => "\[$(x)\]";
+     select_end => "\[.*\]";
+     select_end_match_eof => "false";
+     }
+```
+
+If the `select_end` pattrent was not matched and `select_end_match_eof` is set to true (or skipped), the end of file will
```
Comment (nickanderson) [inline reference/promise-types/edit_line.markdown]: small typo pattrent should be pattern

File: reference/promise-types/edit_line.markdown [hunk 2]
```diff
@@ -411,7 +412,37 @@ mark out the region of a file to be edited.
...
-to the end of the file.
+By default end of the file will be used as end of the region if `select_end` pattern was not 
+found inside the file. If you don't want the end of the file to be implicitly matched as the end of the region, you can set [`select_end_match_eof`][bundle edit_line#select_end_match_eof] 
+value to false (true by default). 
+If the `select_end` promise will be ommited, the selected region will run to the end of the file no matter, what the value of `select_end_match_eof` is set to.
+
+
+#### select_end_match_eof
+
+**Description:** [Anchored][anchored] Whether to match end of file as the end on the region if `select_end` was not matched for the given region
```
Comment (nickanderson) [inline reference/promise-types/edit_line.markdown]: Anchored? This isn't a regex is it?

File: reference/promise-types/edit_line.markdown [hunk 3]
```diff
@@ -411,7 +412,37 @@ mark out the region of a file to be edited.
...
 ```
 
-If you want to match from a starting location to the end of the file
-(even if there are other lines matching `select_start` intervening),
-then just omit the `select_end` promise and the selected region will run
-to the end of the file.
+By default end of the file will be used as end of the region if `select_end` pattern was not 
+found inside the file. If you don't want the end of the file to be implicitly matched as the end of the region, you can set [`select_end_match_eof`][bundle edit_line#select_end_match_eof] 
+value to false (true by default). 
+If the `select_end` promise will be ommited, the selected region will run to the end of the file no matter, what the value of `select_end_match_eof` is set to.
```
Comment (estenberg) [inline reference/promise-types/edit_line.markdown]: Would change this sentence to "If the `select_end` attribute is omitted, the selected region will run to the end of the file no matter what the value of `select_end_match_eof` is set to."

File: reference/promise-types/edit_line.markdown [hunk 4]
```diff
@@ -411,7 +412,37 @@ mark out the region of a file to be edited.
      }
 ```
 
-If you want to match from a starting location to the end of the file
-(even if there are other lines matching `select_start` intervening),
-then just omit the `select_end` promise and the selected region will run
-to the end of the file.
+By default end of the file will be used as end of the region if `select_end` pattern was not
```
Comment (estenberg) [inline reference/promise-types/edit_line.markdown]: Would change to "By default, the end of the file is used as end of region if `select_end` does not match anything."

File: reference/promise-types/edit_line.markdown [hunk 5]
```diff
@@ -411,7 +412,37 @@ mark out the region of a file to be edited.
      }
 ```
 
-If you want to match from a starting location to the end of the file
-(even if there are other lines matching `select_start` intervening),
-then just omit the `select_end` promise and the selected region will run
-to the end of the file.
+By default end of the file will be used as end of the region if `select_end` pattern was not 
+found inside the file. If you don't want the end of the file to be implicitly matched as the end of the region, you can set [`select_end_match_eof`][bundle edit_line#select_end_match_eof]
```
Comment (estenberg) [inline reference/promise-types/edit_line.markdown]: ...you can set [`select_end_match_eof`][bundle edit_line#select_end_match_eof] to false..

**Comment by estenberg** [comment]
Looks good to me except my comments.

File: reference/promise-types/edit_line.markdown (hunk 1 above)
Comment (pasinskim) [inline reference/promise-types/edit_line.markdown]: Thanks, fixing!

File: reference/promise-types/edit_line.markdown (hunk 2 above)
Comment (pasinskim) [inline reference/promise-types/edit_line.markdown]: Fixing!

**Comment by pasinskim** [comment]
Can you recheck @estenberg @nickanderson. We've changed default so the documentation was changed as well.

File: reference/promise-types/edit_line.markdown [hunk 6]
```diff
@@ -410,8 +411,37 @@ mark out the region of a file to be edited.
...
 If you want to match from a starting location to the end of the file
 (even if there are other lines matching `select_start` intervening),
 then just omit the `select_end` promise and the selected region will run
 to the end of the file.
+The default behavior can be modified by using  [`select_end_match_eof`][bundle edit_line#select_end_match_eof]. 
+
+#### select_end_match_eof
+
+**Description:** Whether to match end of file as the end on the region if `select_end` was not matched for the given region.
+If `select_end_match_eof` is true, the selected region will run to the end of the file.
```
Comment (nickanderson) [inline reference/promise-types/edit_line.markdown]: I think I would simplify the wording a bit.

When `select_end_match_eof` is set to true `select_end` will consider end of file as the end region if it is unable to match.

**Comment by pasinskim** [comment]
All issues fixed!

**Comment by eysteins** [comment]
Looks great.
